MekHQ/src/mekhq/campaign/personnel/Person.java

```diff
@@ -2654,6 +2652,25 @@ public String toString() {
         return getFullName();
     }
 
+    /**
+     * Two people are determined to be equal if they have the same id
+     * @param object the object to check if it is equal to the person or not
+     * @return true if they have the same id, otherwise false
+     */
+    @Override
+    public boolean equals(Object object) {
+        if (!(object instanceof Person)) {
+            return false;
+        } else {
+            return getId().equals(((Person) object).getId());
```

Comment:
Should have a reference equality check first.

Reply:
Is that for performance reasons?

Reply:
Part of the recommendations for implementing Object::equals.

Reply:
Implemented the fix